### DISCUSSION TITLE
Making filter using input text as date range picker

### DISCUSSION THREAD

**Discussion by hadyjsc** [Okipa/laravel-table#132] [General]
Hi, is it possible to implement filter using input text, i want to use date picker from bootstrap date picker library. I think by add date picker class to input text class we can get the value, and then passing the value to Builder with `$query->whereBetween($this->attribute, [$from, $to]);`. But the existing filter just supported to filter with dropdown html tag.

Thank you.

**Comment by hadyjsc** [comment]
I try to make new filter in existing repo, but i just make for my private project. I hope you can make it more perfect!!, coz i write with laravel less then 2 month. so my part is bad practice for implement and only applies for temporary. 

- [Abstracts/AbstractDatePicker.php](https://pastebin.com/wyWWUFLW)
- [Filters/DatePickerFilter.php](https://pastebin.com/qSv6UeKf)
- [Livewire/Table.php](https://pastebin.com/twLX0edM)
- [Table.php](https://pastebin.com/48qC9yei)
- [Template](https://pastebin.com/svkn4G8R)

Then, i must use the library from my partial template `script.blade.php` like this
```
<livewire:scripts />
<script>
    Livewire.on('datepicker:script', function() {
        $('.daterange-cus').daterangepicker({
            locale: {
                format: 'YYYY-MM-DD'
            },
            autoUpdateInput: false,
        });

        $('.daterange-cus').on('apply.daterangepicker', function(ev, picker) {
            var val = picker.startDate.format('YYYY-MM-DD') + ' - ' + picker.endDate.format('YYYY-MM-DD');
            Livewire.emit('datepicker:script:set', 'date_time_filter', val)
            $(this).val(val);
        });

        $('.daterange-cus').on('cancel.daterangepicker', function(ev, picker) {
            Livewire.emit('datepicker:script:set', 'date_time_filter', '')
            $(this).val('');
        });
    });
</script>
```

**Reply by Okipa** [reply]
Hi and thanks for your suggestion.
Until now, I managed to handle this kind of extra filter out of the package because it would being a lot more complexity into it.

That being said, I'll think about it to check if there would be a way to implement this.

Coming back to you in a few days!

**Reply by hadyjsc** [reply]
Thanks for the effort @Okipa , i hope everything goes smoothly and can be resolved.
